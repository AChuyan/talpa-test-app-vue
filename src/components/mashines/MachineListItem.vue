<template>
  <router-link :to="routerLink" class="machine-list-item">
    <p>{{ title }}</p>
  </router-link>
</template>

<script>
  export default {
    name: 'machine-list-item',
    props: ['machine'],
    computed: {
      routerLink: function () {
        return `/machine/${this.machine.id}`;
      },
      title: function () {
        return this.machine.name.split('_').join(' ');
      }
    }
  }
</script>

<style scoped>
  .machine-list-item {
    color: #032036;
    text-decoration: none;
  }

  .machine-list-item:hover {
    color: #344974;
    text-decoration: underline;
  }

  .machine-list-item:visited {
    color: #032036;
  }

  .machine-list-item > p {
    font: 500 1rem 'Avenir', Helvetica, Arial, sans-serif;
    text-transform: capitalize;
    text-align: left;
  }
</style>


