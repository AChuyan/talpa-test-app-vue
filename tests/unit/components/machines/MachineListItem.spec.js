import { shallowMount } from '@vue/test-utils'
import { RouterLinkStub } from '@vue/test-utils';
import MachineListItem from '@/components/machines/MachineListItem.vue'

describe('MachineListItem component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(MachineListItem, {
      methods: {},
      mocks: {},
      propsData: {
        machine: { id: 'id', name: 'name' }
      },
      stubs: {
        RouterLink: RouterLinkStub
      },
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
});
