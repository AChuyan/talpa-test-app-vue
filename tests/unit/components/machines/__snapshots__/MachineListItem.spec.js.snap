// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MachineListItem component renders correctly 1`] = `
<a
  class="machine-list-item"
>
  <p>
    name
  </p>
</a>
`;
